File: real-time/file.cs
using System;
using WebSocketSharp;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System.Text;

namespace RealtimeTranscription
{
    class Program
    {
        static void Main(string[] args)
        {
            var authHeader = new string[] { "Authorization: {your_api_token}" };
            var sampleRate = 16000;
            var wordBoost = new string[] { "HackerNews", "Twitter" };
            var url = $"wss://api.assemblyai.com/v2/realtime/ws?word_boost={JsonConvert.SerializeObject(wordBoost)}&sample_rate={sampleRate}";
            var socket = new WebSocket(url, authHeader);

            socket.OnMessage += (sender, e) =>
            {
                var message = JObject.Parse(e.Data);
                if (message["message_type"].ToString() == "SessionBegins")
                {
                    var session_id = message["session_id"].ToString();
                    var expires_at = message["expires_at"].ToString();
                    Console.WriteLine($"Session ID: {session_id}, Expires At: {expires_at}");
                }
                else if (message["message_type"].ToString() == "PartialTranscript")
                {
                    var session_id = message["session_id"].ToString();
                    var audio_start = message["audio_start"].ToObject<int>();
                    var audio_end = message["audio_end"].ToObject<int>();
                    var confidence = message["confidence"].ToObject<float>();
                    var text = message["text"].ToString();
                    Console.WriteLine($"Session ID: {session_id}, Audio Start: {audio_start}, Audio End: {audio_end}, Confidence: {confidence}, Text: {text}");
                }
                else if (message["message_type"].ToString() == "Transcript")
                {
                    var session_id = message["session_id"].ToString();
                    var audio_start = message["audio_start"].ToObject<int>();
                    var audio_end = message["audio_end"].ToObject<int>();
                    var confidence = message["confidence"].ToObject<float>();
                    var text = message["text"].ToString();
                    Console.WriteLine($"Session ID: {session_id}, Audio Start: {audio_start}, Audio End: {audio_end}, Confidence: {confidence}, Text: {text}");
                }
            };

            socket.OnError += (sender, e) =>
            {
                var errorObject = JObject.Parse(e.Message);
                if (errorObject["error_code"].ToString() == "4031")
                {
                    Console.WriteLine("Session idle for too long");
                }
            };

            socket.OnClose += (sender, e) =>
            {
                Console.WriteLine("WebSocket closed");
            };

            socket.Connect();
        }

        static void SendAudio(WebSocket socket, byte[] audioData)
        {
            var payload = new
            {
                audio_data = Convert.ToBase64String(audioData),
            };
            socket.Send(JsonConvert.SerializeObject(payload));
        }

        static void TerminateSession(WebSocket socket)
        {
            var payload = new { terminate_session = true };
            var message = JsonConvert.SerializeObject(payload);
            socket.Send(message);
            socket.Close();
        }
    }
}

File: real-time/file.php
<?php

// Make sure to include a WebSocket client library like Textalk/websocket-php by running `composer require textalk/websocket`.
require 'vendor/autoload.php'; 

use WebSocket\Client;

$authHeader = array(
    "Authorization: {your_api_token}"
);

$sampleRate = 16000;
$wordBoost = array("HackerNews", "Twitter");
$wordBoostParam = urlencode(json_encode($wordBoost));
$url = "wss://api.assemblyai.com/v2/realtime/ws?word_boost={$wordBoostParam}&sample_rate={$sampleRate}";
$socket = new Client($url, array('headers' => $authHeader));

$socket->on('message', function ($data) {
    $message = json_decode($data, true);
    if ($message['message_type'] === 'SessionBegins') {
        $session_id = $message['session_id'];
        $expires_at = $message['expires_at'];
        echo "Session ID: $session_id\n";
        echo "Expires at: $expires_at\n";
    } elseif ($message["message_type"] === "PartialTranscript") {
        echo "Partial transcript received: " . $message["text"] . "\n";
    } elseif ($message['message_type'] === 'FinalTranscript') {
        echo "Final transcript received: " . $message["text"] . "\n";
    }
});

$socket->on('error', function ($error) {
    $errorObject = json_decode($error, true);
    if ($errorObject['error_code'] === 4001) {
        echo "Not Authorized";
    }
});

$socket->on('close', function () {
    echo "WebSocket closed";
});

function send_audio($socket, $audio_data) {
    $payload = array(
        "audio_data" => base64_encode($audio_data)
    );
    $socket->send(json_encode($payload));
}

function terminateSession($socket) {
    $payload = array("terminate_session" => true);
    $message = json_encode($payload);
    $socket->send($message);
    $socket->close();
}
